sklearn/metrics/tests/test_classification.py

```diff
@@ -914,6 +914,18 @@ def test_confusion_matrix_multiclass_subset_labels():
                          labels=[extra_label, extra_label + 1])
 
 
+@pytest.mark.parametrize(
+    'labels', (None, [0, 1], [0, 1, 2]),
+    ids=['None', 'binary', 'multiclass']
+)
+def test_confusion_matrix_on_zero_length_input(labels):
+    labels = [] if not labels else labels
```

Comment:
```suggestion
```

Reply:
So I think I missed a bit of the point when I tried to fix up my own test for the `labels=None` case.  `labels=None` is a special case that is distinct from `labels=[]` (which is a case that we may not want to test at all if we don't want to explicitly support that usage).  I'll correct that case so that it actually tests what it is mean to test.